src/realtime/pubsub/consumer_manager.rs

```diff
@@ -31,6 +30,9 @@ pub enum Event<'l> {
     DeleteMessage { id: &'l str },
     /// A user started typing in a channel.
     Typing { user: &'l PublicUserInfo },
+    /// Either an incoming friend request, or someone accepted an outgoing friend request.
+    /// The client should keep track of pending requests.
+    FriendRequestUpdate { user: &'l PublicUserInfo },
```

Comment:
Maybe we should just have `FriendAdded`, `FriendRemoved` and `FriendRequestRemoved` events? Makes it easier on the client and clearer.

Reply:
Would we need separate events for an incoming friend request cancelled and an outgoing friend request being denied?

Reply:
An incoming request being rejected is almost always a user-initiated action, (eg. clicking the "deny friend request" button) so I don't think we need a realtime event for it at all, just one for getting an outgoing request rejected by the other party.

Reply:
I mean as in someone sending you a friend request and then retracting that friend request. I suppose we should use the same event for that and someone denying your request as the api endpoints prevent that kind of duplication. See https://github.com/zlingapp/server/pull/21/commits/0b1ba1cf1190770b59e0e488adb7ef5ee1bd86e4